src/session/client-certificate.c

```diff
@@ -328,8 +377,37 @@ cockpit_session_client_certificate_map_user (const char *client_certificate_file
       return NULL;
     }
 
+  /* check if stdin is a Unix socket; then we are being called via cockpit-session@.service and need to
```

Comment:
Do we use a pipe for the direct-spawn case?  I forget and/or think we changed this at some point?

Reply:
Isn't passing our activating socket on stdin considered by the systemd folks to be a bit gauche?  I recall reading somewhere that they'd prefer we use fd 3 and an environment variable, or something...

Reply:
> Do we use a pipe for the direct-spawn case? I

Yes, `g_spawn_async_with_pipes()` in   `session_start_process()` (unless it's an unix socket path).